src/services/middlewareServices/AuthenticationMiddlewareService.ts

```diff
@@ -10,63 +11,141 @@ import { AuthError } from "@errors/AuthError"
 
 import jwtUtils from "@utils/jwt-utils"
 
-import { E2E_TEST_EMAIL, E2E_ISOMER_ID } from "@root/constants"
+import { SessionDataProps } from "@root/classes"
+import {
+  E2E_TEST_EMAIL,
+  E2E_ISOMER_ID,
+  E2E_EMAIL_ADMIN_ISOMER_ID,
+  E2E_EMAIL_COLLAB_ISOMER_ID,
+} from "@root/constants"
 import { BadRequestError } from "@root/errors/BadRequestError"
 import { SessionData } from "@root/types/express/session"
 
-const E2E_TEST_REPO = config.get("cypress.e2eTestRepo")
+export const E2E_TEST_REPO = config.get("cypress.e2eTestRepo")
+export const E2E_EMAIL_TEST_REPO = config.get("cypress.e2eEmailTestRepo")
 const E2E_TEST_SECRET = config.get("cypress.e2eTestSecret")
-const E2E_TEST_GH_TOKEN = config.get("cypress.e2eTestGithubToken")
-const E2E_TEST_USER = "e2e-test"
+
+export const E2E_TEST_GH_TOKEN = config.get("cypress.e2eTestGithubToken")
+export const E2E_TEST_USER = "e2e-test"
 const GENERAL_ACCESS_PATHS = [
   "/v1/sites",
   "/v1/auth/whoami",
   "/v2/sites",
   "/v2/auth/whoami",
 ]
 
-type VerifyAccessProps = SessionData & {
-  cookies: {
-    isomercms: string
-    isomercmsE2E?: string
-  }
+type UnverifiedSession =
+  | Partial<SessionData>
+  | { userInfo: Record<string, never> }
+  | { userInfo: SetOptional<SessionData["userInfo"], "email"> }
+
+export type VerifyAccessProps = UnverifiedSession & {
+  // NOTE: Either both properties are present on the cookie
+  // or none are present.
+  // We disallow having 1 or the other.
+  cookies: RequireAllOrNone<
+    {
+      isomercmsE2E: string
+      e2eUserType: string
+    },
+    "e2eUserType" | "isomercmsE2E"
+  >
   url: string
 }
 
+type TestUserTypes = "Email admin" | "Email collaborator" | "Github user"
```

Comment:
Nit: would an enum here make it easier to ensure we're referencing the right value?

Reply:
hmm +1 for not using enums in ts tho :(

Reply:
hmm, i can do a `const` object + type manipulation bah if we just want a nice aliasing model like `UserTypes.Email.Admin`

Reply:
Yea that would be good